### PR TITLE
Standardize CI 20200203

### PR DESCRIPTION
**Pull request by phil-davis**
- cleanup `.gitignore`
- implement Travis like in `skel` (separate `phpstan` job, `--prefer-lowest` in matrix)
- apply `php-cs-fixer` changes for `tests` folder
- move tests into `tests/Cache` and adjust composer `autoload-dev` section to fix phpstan:
```
composer phpstan
> @php vendor/bin/phpstan analyse lib tests
Note: Using configuration file /home/phil/git/sabre-io/cache/phpstan.neon.
 10/10 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ------------------------------------------------------------------- 
  Line   tests/Cache/AbstractCacheTest.php                                  
 ------ ------------------------------------------------------------------- 
         Class Sabre\Cache\AbstractCacheTest was not found while trying to  
         analyse it - autoloading is probably not configured properly.      
 ------ ------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------- 
  Line   tests/Cache/ApcuTest.php                                               
 ------ ----------------------------------------------------------------------- 
         Class Sabre\Cache\ApcuTest was not found while trying to analyse it -  
         autoloading is probably not configured properly.                       
 ------ ----------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------- 
  Line   tests/Cache/MemcachedTest.php                                          
 ------ ----------------------------------------------------------------------- 
         Class Sabre\Cache\MemcachedTest was not found while trying to analyse  
         it - autoloading is probably not configured properly.                  
 ------ ----------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------- 
  Line   tests/Cache/MemoryCacheTest.php                                  
 ------ ----------------------------------------------------------------- 
         Class Sabre\Cache\MemoryCacheTest was not found while trying to  
         analyse it - autoloading is probably not configured properly.    
 ------ ----------------------------------------------------------------- 

 [ERROR] Found 4 errors                                                         

Script @php vendor/bin/phpstan analyse lib tests handling the phpstan event returned with error code 1
````
(I made it the similar pattern to what happens in `vobject` )

### CODE REVIEW

File: tests/Cache/ApcuTest.php [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
+    {
         if (!function_exists('apcu_store')) {
             $this->markTestSkipped('Apcu extension is not loaded');
         }
         if (!ini_get('apc.enabled')) {
             $this->markTestSkipped('apc.enabled is set to 0. Enable it via php.ini');
         }
 
-        if (php_sapi_name() === 'cli' && !ini_get('apc.enable_cli')) {
+        if ('cli' === php_sapi_name() && !ini_get('apc.enable_cli')) {
```
Comment (phil-davis) [inline tests/Cache/ApcuTest.php]: It likes yoda style

File: tests/Cache/AbstractCacheTest.php [hunk 2]
```diff
@@ -268,7 +250,7 @@ function testSetMultipleExpireDateIntervalNotExpired() {
 
         $count = 0;
         foreach ($result as $key => $value) {
-            $count++;
+            ++$count;
```
Comment (phil-davis) [inline tests/Cache/AbstractCacheTest.php]: The fixer likes to put `++` before - in this context it makes no difference.

**Comment by staabm** [comment]
We should have this included https://github.com/sabre-io/dav/pull/1237 (also in the skel repo)

**Comment by phil-davis** [comment]
`skel` repo has this stuff, except that it does not actually have any tests, so it does not have an example of `autoload-dev` things.

**Comment by staabm** [comment]
My comment was not about autoloading but about not building unnecessary branches

**Pull request merged**